### PR TITLE
Update Diagnosis and Notification to return proper date

### PR DESCRIPTION
**Pull request by proteusvacuum**
@esoergel 
This will fix the issue where all BETSDiagnosisAndNotificationRepeater EventDates are getting set to today's date.

🐟 

TODO:
- [x] Add this property to the app (once domain migration is complete)
- [ ] Run management command to update past cases

cc: @gcapalbo

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/util.py [hunk 1]
```diff
@@ -177,3 +177,35 @@ def iter_cases(case_ids, strip_history=False, wrap=True):
     else:
         for case in CommCareCase.bulk_get_lite(case_ids, wrap=wrap):
             yield case
+
+
+def get_datetime_case_property_changed(case, case_property_name, value):
```
Comment (proteusvacuum) [inline corehq/ex-submodules/casexml/apps/case/util.py]: wasn't sure where the best place for this util to live was

File: corehq/ex-submodules/casexml/apps/case/util.py [hunk 2]
```diff
@@ -177,3 +177,35 @@ def iter_cases(case_ids, strip_history=False, wrap=True):
     else:
         for case in CommCareCase.bulk_get_lite(case_ids, wrap=wrap):
             yield case
+
+
+def get_date_case_property_changed(case, case_property_name, value):
+    """Returns the date a particular case property was changed to a specific value
```
Comment (esoergel) [inline corehq/ex-submodules/casexml/apps/case/util.py]: Might want to clarify that this gives the _first_ date a particular property was set to particular value.

File: custom/enikshay/management/commands/base.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        errors = []
+
+        case_ids = get_all_episode_ids(domain)
+        cases = iter_all_active_person_episode_cases(domain, case_ids)
+
+        for person, episode in with_progress_bar(cases, len(case_ids)):
+            update_json = {}
+            for updater in self.updaters:
+                try:
+                    update_json.update(updater(self.domain, episode).update_json())
```
Comment (esoergel) [inline custom/enikshay/management/commands/base.py]: I heard you like updates, so I'll update your update with the update from an update.

File: custom/enikshay/management/commands/base.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                except Exception as e:
+                    error = [episode.case_id, episode.domain, updater.__name__, e]
+                    errors.append(error)
+                    print "{}: {} - {}".format(*error)
```
Comment (esoergel) [inline custom/enikshay/management/commands/base.py]: Were you planning on dumping these to a file or something?

File: custom/enikshay/management/commands/base.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+from django.core.management.base import BaseCommand
+
+from corehq.apps.hqcase.utils import bulk_update_cases
+from corehq.util.log import with_progress_bar
+
+from custom.enikshay.case_utils import get_all_episode_ids, iter_all_active_person_episode_cases
+
+
+class ENikshayBatchCaseUpdaterCommand(BaseCommand):
```
Comment (esoergel) [inline custom/enikshay/management/commands/base.py]: I like this idea

File: custom/enikshay/integrations/bets/repeater_generators.py [hunk 6]
```diff
@@ -216,7 +216,7 @@ def create_diagnosis_and_notification_payload(cls, episode_case):
 
         return cls(
             EventID=DIAGNOSIS_AND_NOTIFICATION_EVENT,
-            EventOccurDate=cls._india_now(),
+            EventOccurDate=episode_case.get_case_property('bets_first_prescription_voucher_redeemed_date'),
```
Comment (esoergel) [inline custom/enikshay/integrations/bets/repeater_generators.py]: :tada:

**Review by esoergel — APPROVED** [review]

**Review by esoergel — APPROVED** [review]

**Pull request merged**